hydroflow_plus/src/stream.rs

```diff
@@ -19,38 +19,40 @@ use crate::ir::{DebugInstantiate, HfPlusLeaf, HfPlusNode, HfPlusSource};
 use crate::location::{CanSend, Location, LocationId};
 use crate::Cluster;
 
-/// Marks the stream as being asynchronous, which means the presence
-/// of all elements is directly influenced by the runtime's batching
-/// behavior. Aggregation operations are not permitted on streams
-/// with this tag because the developer has not explicitly specified
-/// if they want to aggregate over the entire stream or just the
-/// current batch.
-pub struct Async {}
-
-/// Marks the stream as being windowed, which means the developer has
-/// opted-into either a batched or persistent windowing semantics.
-/// Aggregation operations are permitted on streams with this tag.
-pub struct Windowed {}
+/// Marks the stream as being unbounded, which means that it is not
+/// guaranteed to be complete in finite time.
+pub struct Unbounded {}
+
+/// Marks the stream as being bounded, which means that it is guaranteed
+/// to be complete in finite time.
+pub struct Bounded {}
+
+/// Marks the stream as existing outside of a clock domain.
+pub struct NoTick {}
+/// Marks the stream as being inside the single global clock domain.
+pub struct Tick {}
```

Comment:
(nit: for marker types, better to use empty enums `pub enum Tick {}` since they can't be instantiated)